File: anagrams/anagrams.py
def anagrams(s1, s2):
  dict = {}
  for char in s1:
    if char in dict:
      dict[char] += 1
    else:
      dict[char] = 1
  
  for char in s2:
    if char not in dict:
      return False
    elif dict[char] == 0:
      return False
    
    dict[char] -= 1
  
  for val in dict.values():
    if val != 0: return False
  
  return True

File: anagrams/anagrams.js
const anagrams = (s1, s2) => {
  const hashMap = {}
  for (let i = 0; i < s1.length; i++) {
    const letter = s1[i];
    hashMap[letter] ? hashMap[letter]++ : hashMap[letter] = 1
  }
  
  for (let i = 0; i < s2.length; i++) {
    const letter = s2[i];
    
    if (!hashMap[letter] || hashMap[letter] === 0) return false;
    hashMap[letter]--;
  }
  
  for (const letter in hashMap) {
    if (hashMap[letter] !== 0) return false
  }
  return true
};
​
module.exports = {
  anagrams,
};
​